### ISSUE TITLE
Using ObjectToWorld3x4() with -emit-spirv-directly causes "End of input reached while decoding OpLoad"

### ISSUE THREAD

**Issue by natevm** [shader-slang/slang#3422]
I have a small reproducer here: https://gitlab-master.nvidia.com/nmorrical/object-to-world-bug-repro

I'm using the -emit-spirv-directly flag, and calling "ObjectToWorld3x4()" inside a closest hit program. If this function is present in the closest hit program, slangc.exe crashes.

Using the transpose function, "ObjectToWorld4x3()" works without issue.

Here is the specific shader code involved: https://gitlab-master.nvidia.com/nmorrical/object-to-world-bug-repro/-/blob/main/repro.slang?ref_type=heads

```
[shader("closesthit")]
void TriClosestHit(inout Payload payload, in float2 bc) {
    float3 color = float3(1.f - bc.x - bc.y, bc.x, bc.y);

    // This causes the driver to crash...
    float3x4 transform = ObjectToWorld3x4();
    payload.color = mul(transform, float4(color, 1.f));

    // But this works just fine
    // float4x3 transform = ObjectToWorld4x3();
    // payload.color = mul(float4(color, 1.f), transform);
}
```

The generated SPIRV looks like this:

```
; SPIR-V
; Version: 1.5
; Generator: Khronos; 40
; Bound: 34
; Schema: 0
               OpCapability RayTracingKHR
               OpCapability Shader
               OpExtension "SPV_KHR_ray_tracing"
               OpMemoryModel Logical GLSL450
               OpEntryPoint ClosestHitNV %TriClosestHit "TriClosestHit" %27 %9 %20
               OpSource Unknown 1
               OpName %mat "mat"
               OpName %Payload "Payload"
               OpMemberName %Payload 0 "color"
               OpName %TriClosestHit "TriClosestHit"
               OpDecorate %20 BuiltIn ObjectToWorldNV
               OpMemberDecorate %Payload 0 Offset 0
       %void = OpTypeVoid
          %3 = OpTypeFunction %void
      %float = OpTypeFloat 32
    %v2float = OpTypeVector %float 2
%_ptr_HitAttributeNV_v2float = OpTypePointer HitAttributeNV %v2float
    %float_1 = OpConstant %float 1
    %v3float = OpTypeVector %float 3
%mat4v3float = OpTypeMatrix %v3float 4
%_ptr_Input_mat4v3float = OpTypePointer Input %mat4v3float
    %v4float = OpTypeVector %float 4
%mat3v4float = OpTypeMatrix %v4float 3
    %Payload = OpTypeStruct %v3float
%_ptr_IncomingRayPayloadNV_Payload = OpTypePointer IncomingRayPayloadNV %Payload
        %int = OpTypeInt 32 1
      %int_0 = OpConstant %int 0
%_ptr_IncomingRayPayloadNV_v3float = OpTypePointer IncomingRayPayloadNV %v3float
          %9 = OpVariable %_ptr_HitAttributeNV_v2float HitAttributeNV
         %20 = OpVariable %_ptr_Input_mat4v3float Input
         %27 = OpVariable %_ptr_IncomingRayPayloadNV_Payload IncomingRayPayloadNV
%TriClosestHit = OpFunction %void None %3
          %4 = OpLabel
          %7 = OpLoad %v2float %9
         %10 = OpCompositeExtract %float %7 0
         %11 = OpFSub %float %float_1 %10
         %13 = OpLoad %v2float %9
         %14 = OpCompositeExtract %float %13 1
         %15 = OpFSub %float %11 %14
         %17 = OpCompositeConstruct %v3float %15 %10 %14
error: 41: End of input reached while decoding OpLoad starting at word 170: expected more operands after 3 words.
```

**Comment by pmistryNV** [comment]
Looking into the bug

**Comment by pmistryNV** [comment]
@natevm kindly confirm and I will close the bug

**Comment by natevm** [comment]
Works for me! We should check to see if this also fixes issue #3359
